src/sql/parts/common/customInputConverter.ts

```diff
@@ -59,20 +57,23 @@ export function convertEditorInput(input: EditorInput, options: IQueryEditorOpti
 
 		//Notebook
 		let notebookValidator = instantiationService.createInstance(NotebookInputValidator);
-		uri = getNotebookEditorUri(input);
+		uri = getNotebookEditorUri(input, instantiationService);
 		if(uri && notebookValidator.isNotebookEnabled()){
-			//TODO: We need to pass in notebook data either through notebook input or notebook service
-			let fileName: string = 'untitled';
-			let providerId: string = DEFAULT_NOTEBOOK_PROVIDER;
-			if (input) {
-				fileName = input.getName();
-				providerId = getProviderForFileName(fileName);
-			}
-			let notebookInputModel = new NotebookInputModel(uri, undefined, false, undefined);
-			notebookInputModel.providerId = providerId;
-			//TO DO: Second parameter has to be the content.
-			let notebookInput: NotebookInput = instantiationService.createInstance(NotebookInput, fileName, notebookInputModel);
-			return notebookInput;
+			return withService<INotebookService, NotebookInput>(instantiationService, INotebookService, notebookService => {
+
+				//TODO: We need to pass in notebook data either through notebook input or notebook service
+				let fileName: string = 'untitled';
+				let providerId: string = DEFAULT_NOTEBOOK_PROVIDER;
+				if (input) {
+					fileName = input.getName();
+					providerId = getProviderForFileName(fileName, notebookService);
+				}
+				let notebookInputModel = new NotebookInputModel(uri, undefined, false, undefined);
+				notebookInputModel.providerId = providerId;
+				//TO DO: Second parameter has to be the content.
```

Comment:
Can you clarify this? the 2nd parameter that I see below is fileName

Reply:
Great question. I just copied this code (see it's in the red section too). I think we can just remove the comments